src/envoy/auth/jwt.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef PROXY_JWT_H
+#define PROXY_JWT_H
+
+#include "openssl/evp.h"
+#include "rapidjson/document.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+class Jwt {
+ public:
+  static std::unique_ptr<rapidjson::Document> decode(
```

Comment:
We might need to eventually convert this to a class, public key as member. That should be in follow up PRs.